src/me/john000708/barrels/Barrels.java

```diff
@@ -149,8 +152,15 @@ public boolean onRightClick(ItemUseEvent itemUseEvent, Player player, ItemStack
                     Block clickedBlock = itemUseEvent.getClickedBlock();
                     if (BlockStorage.getLocationInfo(clickedBlock.getLocation(), "explosion") == null) {
                         BlockStorage.addBlockInfo(clickedBlock, "explosion", "true");
-                        player.getInventory().setItem(player.getInventory().getHeldItemSlot(), InvUtils.decreaseItem(itemStack, 1));
+                        // Fixes issue #6.
+                        //player.getInventory().setItem(player.getInventory().getHeldItemSlot(), InvUtils.decreaseItem(itemStack, 1));
+                        int amount = itemStack.getAmount();
+                        if (amount <= 1) {
+                            itemStack.setAmount(0);
+                        }
+                        else itemStack.setAmount(amount - 1);
                         player.sendMessage(ChatColor.GREEN + "Module successfully applied!");
+                        return true;
```

Comment:
Why was there no return inbefore...????

Reply:
inbefore?

Reply:
Before your commit, this return true; statement should have been there all along.